### PR TITLE
fix: several bugs due to rebase

### PR DESCRIPTION
**Pull request by kittybest**
**Description**
- fix missing return types
- adding empty line between elements
- re-order imports
- re-order functions and components in a single file
- remove redundant conditional check
- fix all lint errors

### CODE REVIEW

File: public/mockServiceWorker.js [hunk 1]
```diff
@@ -1,11 +1,14 @@
+/* eslint-disable */
```
Comment (ctrlc03) [inline public/mockServiceWorker.js:1]: this is inserted automatically right? if not can we remove it?

**Review by ctrlc03 — COMMENTED** [review]
Thanks so much for this! just one comment

File: public/mockServiceWorker.js (hunk 1 above)
Comment (kittybest) [reply public/mockServiceWorker.js:1]: let me try

File: src/components/Footer.tsx [hunk 2]
```diff
@@ -1,48 +1,48 @@
...
+      <a aria-label="discord" href="https://discord.com/invite/sF5CT5rzrR" rel="noreferrer" target="_blank">
+        <FaDiscord />
+      </a>
+    </div>
+
+    <div className="flex justify-end gap-4">
+      <a className="flex items-center gap-1" href="https://maci.pse.dev" rel="noreferrer" target="_blank">
+        <span>Documentation</span>
+
+        <Image alt="" height="18" src="/arrow-go-to.svg" width="18" />
```
Comment (0xmad) [inline src/components/Footer.tsx:36]: alt should not be empty (related to all images for this PR)

File: src/components/SortByDropdown.tsx [hunk 3]
```diff
@@ -16,53 +16,47 @@ interface IRadioItemProps {
...
 
-      <DropdownMenu.Portal>
-        <DropdownMenu.Content
-          className="z-50 w-[200px] rounded-md border border-gray-300 bg-white p-2 dark:border-gray-700 dark:bg-gray-900 dark:text-white"
-          sideOffset={5}
+        <DropdownMenu.RadioGroup
+          value={value}
+          onValueChange={(v) => {
+            onChange(v);
+          }}
```
Comment (0xmad) [inline src/components/SortByDropdown.tsx:53]: Should be used with `useCallback`

File: src/components/ui/Dialog.tsx [hunk 4]
```diff
@@ -24,60 +27,58 @@ const Content = createComponent(
   }),
 );
 
+interface IDialogProps {
+  title: string | ReactNode;
+  description: string | ReactNode;
```
Comment (0xmad) [inline src/components/ui/Dialog.tsx:32]: ReactNode already has string

File: src/components/ui/Dialog.tsx [hunk 5]
```diff
@@ -24,60 +27,58 @@ const Content = createComponent(
...
-                ></IconButton>
-              </RadixDialog.Close>
-            ) : null}
-          </Content>
-        </div>
-      </RadixDialog.Portal>
-    </RadixDialog.Root>
-  );
-};
+}: IDialogProps & PropsWithChildren): JSX.Element => (
```
Comment (0xmad) [inline src/components/ui/Dialog.tsx:53]: It should be used with extending of `PropsWithChildren`
```suggestion
}: IDialogProps): JSX.Element => (
```

File: src/components/ui/Dialog.tsx [hunk 6]
```diff
@@ -24,60 +27,58 @@ const Content = createComponent(
...
+            <Button size="auto" variant={button} onClick={buttonAction}>
+              {buttonName}
+            </Button>
+          )}
+
+          {onOpenChange ? (
+            <RadixDialog.Close asChild>
+              <IconButton className="absolute right-4 top-4" icon={X} variant="ghost" />
+            </RadixDialog.Close>
+          ) : null}
```
Comment (0xmad) [inline src/components/ui/Dialog.tsx:79]: Do we need to use onOpenChange callback here when closing?

File: src/components/ui/Navigator.tsx [hunk 7]
```diff
@@ -1,19 +1,19 @@
...
-        <Link href="/projects">Projects</Link>
-      </span>
-      <span>{">"}</span>
-      <span className="text-blue-400">
-        <b>{projectName}</b>
-      </span>
-    </div>
-  );
-};
+export const MyNavigator = ({ projectName }: INavigatorProps): JSX.Element => (
```
Comment (0xmad) [inline src/components/ui/Navigator.tsx:7]: Why is it called `MyNavigator`?

File: src/components/ui/Notification.tsx [hunk 8]
```diff
@@ -26,21 +26,21 @@ interface NotificationProps {
   title?: string;
 }
 
-export const Notification = ({
+export const MyNotification = ({
```
Comment (0xmad) [inline src/components/ui/Notification.tsx:29]: The same here with `My` prefix

File: src/contexts/Maci.tsx [hunk 9]
```diff
@@ -232,8 +232,6 @@ export const MaciProvider: React.FC<MaciProviderProps> = ({ children }: MaciProv
       return;
     }
 
-    setIsLoading(true);
```
Comment (0xmad) [inline src/contexts/Maci.tsx:235]: Why did you remove loading logic?

File: src/features/ballot/components/SubmitBallotButton.tsx [hunk 10]
```diff
@@ -1,38 +1,46 @@
...
 
-export const SubmitBallotButton = () => {
+export const SubmitBallotButton = (): JSX.Element => {
   const router = useRouter();
   const [isOpen, setOpen] = useState(false);
   const { onVote, isLoading } = useMaci();
   const { ballot, publishBallot } = useBallot();
   const projectIndices = useProjectIdMapping(ballot);
 
+  const onVotingError = () => {
```
Comment (0xmad) [inline src/features/ballot/components/SubmitBallotButton.tsx:18]: useCallback

File: src/features/projects/components/Projects.tsx [hunk 11]
```diff
@@ -38,46 +40,50 @@ export const Projects = (): JSX.Element => {
...
+            className={clsx("relative", { "animate-pulse": isLoading })}
+            href={`/projects/${item.id}`}
+          >
+            {!results.isLoading && appState === EAppState.RESULTS ? (
+              <ProjectItemAwarded amount={results.data?.projects[item.id]?.votes} />
+            ) : null}
+
+            <ProjectItem
+              action={(e: Event) => {
+                handleAction(e, item.id);
```
Comment (0xmad) [inline src/features/projects/components/Projects.tsx:79]: useCallback

File: src/features/projects/components/ProjectsResults.tsx [hunk 12]
```diff
@@ -24,7 +28,12 @@ export const ProjectsResults = (): JSX.Element => {
             <ProjectItemAwarded amount={results.data?.projects[item.id]?.votes} />
           ) : null}
 
-          <ProjectItem attestation={item} isLoading={isLoading} />
+          <ProjectItem
+            action={() => router.push(`/projects/${item.id}`)}
```
Comment (0xmad) [inline src/features/projects/components/ProjectsResults.tsx:32]: useCallback

File: src/pages/ballot/index.tsx [hunk 13]
```diff
@@ -34,20 +34,22 @@ const ClearBallot = () => {
     <>
       <div
         className="cursor-pointer text-gray-400 underline hover:text-black"
-        onClick={() => setOpen(true)}
+        onClick={() => {
```
Comment (0xmad) [inline src/pages/ballot/index.tsx:37]: useCallback and don't use click handlers with div (should be button in this case)

**Review by 0xmad — CHANGES_REQUESTED** [review]
@kittybest thanks, left some comments

File: src/components/ui/Dialog.tsx (hunk 6 above)
Comment (0xmad) [reply src/components/ui/Dialog.tsx:79]: So, is this dialog closed without any callbacks? Is so, let's keep it as is.

File: src/components/ui/Navigator.tsx (hunk 7 above)
Comment (kittybest) [reply src/components/ui/Navigator.tsx:7]: because the linter keeps telling me `Navigator` is already used as global variable

File: src/components/ui/Notification.tsx (hunk 8 above)
Comment (kittybest) [reply src/components/ui/Notification.tsx:29]: This could be removed

File: src/components/ui/Navigator.tsx (hunk 7 above)
Comment (0xmad) [reply src/components/ui/Navigator.tsx:7]: You can add it to exceptions for linter or give it a different name (for instance Navigation)

File: src/contexts/Maci.tsx (hunk 9 above)
Comment (kittybest) [reply src/contexts/Maci.tsx:235]: because it caused something strange: the `isLoading` changed, affect the exported value, the `useAppState` changed, the `Info` keeps on refreshing.

File: public/mockServiceWorker.js (hunk 1 above)
Comment (kittybest) [reply public/mockServiceWorker.js:1]: I think they are automatically generated, I tried to remove them but then they appear again, how does @0xmad think?

File: src/components/ui/Dialog.tsx (hunk 6 above)
Comment (kittybest) [reply src/components/ui/Dialog.tsx:79]: Yes

File: src/components/ui/Navigator.tsx (hunk 7 above)
Comment (kittybest) [reply src/components/ui/Navigator.tsx:7]: That’s why I call it MyNavigator (as a different name). Would you recommend more to add it to exception or change to another name?

File: src/contexts/Maci.tsx (hunk 9 above)
Comment (0xmad) [reply src/contexts/Maci.tsx:235]: Deleting loading logic is not a fix, it can be something related to render or hook/context dependencies.

File: src/components/ui/Navigator.tsx (hunk 7 above)
Comment (0xmad) [reply src/components/ui/Navigator.tsx:7]: Let's rename it to Navigation

File: src/components/ui/Notification.tsx (hunk 8 above)
Comment (0xmad) [reply src/components/ui/Notification.tsx:29]: I think `Notice` will work, if not, you can add it to exceptions.

Comment (0xmad) [reply src/components/ui/Notification.tsx:29]: And don't forget to rename file (for Navigation too)

File: src/contexts/Maci.tsx (hunk 9 above)
Comment (kittybest) [reply src/contexts/Maci.tsx:235]: So I think the problem is 
https://github.com/privacy-scaling-explorations/maci-rpgf/blob/c280402b2e2ce1513e8aab928a058939d503f13e/src/contexts/Maci.tsx#L264
--> this shows that the `useEffect` hook depends on `votingEndsAt`
however,
https://github.com/privacy-scaling-explorations/maci-rpgf/blob/c280402b2e2ce1513e8aab928a058939d503f13e/src/contexts/Maci.tsx#L87
--> `votingEndsAt` depends on `pollData`
and actually `pollData` is set here
https://github.com/privacy-scaling-explorations/maci-rpgf/blob/c280402b2e2ce1513e8aab928a058939d503f13e/src/contexts/Maci.tsx#L244
which is inside the same `useEffect` hook

So it causes a loop that causing `isLoading` become `true` and `false` so many times, which affect the `useAppState` hook, that also affect the `Info` component.

Comment (kittybest) [reply src/contexts/Maci.tsx]: I added some condition to avoid `setPollData` every time, but not sure if it's a good way. Need some recommendation.

File: src/contexts/Maci.tsx [hunk 14]
```diff
@@ -84,10 +84,10 @@ export const MaciProvider: React.FC<MaciProviderProps> = ({ children }: MaciProv
...
   }, [address, signatureMessage, signMessageAsync, setMaciPrivKey, setMaciPubKey]);
 
-  const votingEndsAt = useMemo(
-    () => (pollData ? new Date(Number(pollData.deployTime) * 1000 + Number(pollData.duration) * 1000) : new Date()),
-    [pollData],
-  );
+  // const votingEndsAt = useMemo(
+  //   () => (pollData ? new Date(Number(pollData.deployTime) * 1000 + Number(pollData.duration) * 1000) : new Date()),
+  //   [pollData],
+  // );
```
Comment (0xmad) [inline src/contexts/Maci.tsx:90]: Let's remove it

File: src/contexts/Maci.tsx [hunk 15]
```diff
@@ -241,10 +241,16 @@ export const MaciProvider: React.FC<MaciProviderProps> = ({ children }: MaciProv
         provider: signer.provider,
       })
         .then((data) => {
-          setPollData(data);
+          if (
+            (pollData && (data.address !== pollData.address || data.isStateAqMerged !== pollData.isStateAqMerged)) ||
```
Comment (0xmad) [inline src/contexts/Maci.tsx:245]: This should be moved to a separate const
```
(data.address !== pollData.address || data.isStateAqMerged !== pollData.isStateAqMerged)
```

File: src/contexts/Maci.tsx (hunk 9 above)
Comment (0xmad) [reply src/contexts/Maci.tsx]: No, it's a good way to prevent it.

File: src/contexts/Maci.tsx (hunk 15 above)
Comment (kittybest) [reply src/contexts/Maci.tsx]: I made this an independent const function, should I just put inside this MACI context? or I think it could be in utils? (not sure if someone else need it)

**Review by ctrlc03 — APPROVED** [review]
Thanks

File: src/features/projects/components/VotingWidget.tsx [hunk 16]
```diff
@@ -33,66 +29,67 @@ export const VotingWidget = ({ projectId }: { projectId: string }) => {
     removeFromBallot(projectId);
     setAmount(undefined);
     setButtonState(0);
-  }, [removeFromBallot]);
+  }, [projectId, removeFromBallot]);
 
-  const handleInput = (e: Event) => {
-    setAmount(e.target?.value as number);
+  const handleInput = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (0xmad) [inline src/features/projects/components/VotingWidget.tsx:34]: Let's use `import type { ChangeEvent } from "react"`

**Review by 0xmad — APPROVED** [review]
@kittybest thanks!
